app/controllers/experiments_controller.rb

```diff
@@ -8,25 +8,37 @@ def show
 
   def new
     @experiment = Experiment.new
+    redirect_to proposal_path(@proposal) unless logged_in? && !@proposal.experiment
   end
 
   def edit
+    redirect_to proposal_experiment_path(@experiment.proposal, @experiment) unless experimentor?(@proposal)
   end
 
   def create
     @experiment = Experiment.new(experiment_params)
-    @experiment.proposal = @proposal
-    if @experiment.save
-      @proposal.status = "in progress"
-      @proposal.save
-      redirect_to @proposal
+    if logged_in? && !@proposal.experiment
+      @proposal.assign_attributes(experimentor_id: current_user.id)
+      @experiment.proposal = @proposal
+      if @experiment.save
+        @proposal.status = "in progress"
+        @proposal.save
+        redirect_to @proposal
+      else
+        render 'new'
+      end
     else
-      render 'new'
+      redirect_to proposal_path(@proposal)
     end
   end
 
   def update
-    if @experiment.update(experiment_params)
+    if experimentor?(@proposal)
+      if @experiment.update(experiment_params)
+        @proposal.update_attributes(status: "closed") if @experiment.results != "" || @experiment.conclusion != ""
+        redirect_to proposal_experiment_path(@experiment.proposal, @experiment)
```

Comment:
if the experimentor?(@proposal) is true but the logic on line 37 is false, what is the case?